tests/DontTest/DontSerialiseTest.php

```diff
@@ -34,12 +35,40 @@ public function nonSerialisableObject() : array
     {
         return [
             [new NotSerialisable()],
+            [new NonDeserialisableImplementingSerializable()],
             [new DontDoIt()],
         ];
     }
 
     public function testSerialisePreventionIsFinal() : void
     {
         self::assertTrue((new \ReflectionMethod(DontSerialise::class, '__sleep'))->isFinal());
+        self::assertTrue((new \ReflectionMethod(DontSerialise::class, 'serialize'))->isFinal());
+        self::assertTrue((new \ReflectionMethod(DontSerialise::class, '__serialize'))->isFinal());
+    }
+
+    public function testManuallyInvokingMethods() : void
+    {
+        $dont = new DontDoIt();
+
+        try {
+            $dont->__serialize();
+            self::markAsFailed('This method must always throw');
```

Comment:
Similar to the above: need `$this->expectException()` and separate tests instead